internal/pkg/endpoints/kubernetes.go

```diff
@@ -511,19 +610,24 @@ func (k *KubernetesTargetRetriever) processEvent(event watch.Event, requireLabel
 		}
 	case watch.Modified:
 		if requireLabel {
-			// If the object requires labels, is scrapable and was not seen before,
-			// we add it.
-			if scrapable && !seen {
+			// If the object requires labels, is scrapable we update it
+			if scrapable {
 				k.addTarget(object, event.Type)
 				return
 			}
 			// If the object is not scrapable and we've seen it before, we remove it.
 			if !scrapable && seen {
 				k.targets.Delete(string(object.GetUID()))
 				debugLogEvent(klog, event.Type, "deleted", object)
+				switch obj := object.(type) {
+				case *apiv1.Service:
+					if e, err := k.client.CoreV1().Endpoints(obj.Namespace).Get(obj.Name, metav1.GetOptions{}); err == nil {
+						k.targets.Delete(string(e.GetUID()))
+						debugLogEvent(klog, event.Type, "deleted", e)
+					}
+				}
 			}
-		}
-		if !requireLabel {
+		} else if !requireLabel {
```

Comment:
```suggestion
		} else {
```

Reply:
didn't you saw "Please, do not try to reduce the amount of code below or simplify the conditionals."?

I'll update that xD